src/main/java/com/massivecraft/factions/data/MemoryBoard.java

```diff
@@ -272,81 +272,84 @@ public int getFactionCoordCountInWorld(Faction faction, String worldName) {
-            }
-            for (int dx = (dz < 3 ? 6 : 3); dx < width; dx++) {
-                if (dx == halfWidth && dz == halfHeight) {
-                    row.then("+").color(ChatColor.AQUA);
-                    if (false) {
-                        row.tooltip(TL.CLAIM_YOUAREHERE.toString());
+        final ArrayList<String> compass =
+            AsciiCompass.getAsciiCompass(inDegrees, ChatColor.RED, FactionsPlugin.getInstance().txt().parse("<a>"));
+
+        final int halfMapWidth = conf.map().getWidth() / 2;
+        final int halfMapHeight = fPlayer.getMapHeight() / 2;
+        final int mapWidth = halfMapWidth * 2 + 1;
+        final int mapHeight = conf.map().isShowFactionKey() ? halfMapHeight * 2 : halfMapHeight * 2 + 1;
+
+        final FLocation topLeft = fLocation.getRelative(-halfMapWidth, -halfMapHeight);
+        final Map<String, Character> factionsList = new HashMap<>();
+
+        int charIndex = 0;
+
+        // ... for each row
+        for (int dz = 0; dz < mapHeight; dz++) {
+            final FancyMessage row = new FancyMessage("");
+
+            // ... add ascii compass
+            if (dz < 3) row.then(compass.get(dz));
+
+            for (int dx = (dz < 3 ? 6 : 3); dx < mapWidth; dx++) {
+                // ... current chunk fplayer is standing in
+                if (dx == halfMapWidth && dz == halfMapHeight) {
+                    row.then("+").color(ChatColor.AQUA).tooltip(TL.CLAIM_YOUAREHERE.toString());
+                    continue;
+                }
+
+                final Faction factionAt = getFactionAt(topLeft.getRelative(dx, dz));
+                final Relation relation = fPlayer.getRelationTo(factionAt);
+
+                if (factionAt.isWilderness()) {
+                    row.then("-").color(conf.colors().factions().getWilderness());
+                } else if (factionAt.isSafeZone()) {
+                    row.then("+").color(conf.colors().factions().getSafezone());
+                } else if (factionAt.isWarZone()) {
+                    row.then("+").color(conf.colors().factions().getWarzone());
+                }
+
+                // ... check relations
+                else if (factionAt == senderFaction || factionAt == senderFactionAtLocation || relation.isAtLeast(Relation.ALLY) ||
+                    conf.map().isShowNeutralFactions() && relation.equals(Relation.NEUTRAL) ||
+                    conf.map().isShowEnemyFactions() && relation.equals(Relation.ENEMY) ||
+                    conf.map().isShowTruceFactions() && relation.equals(Relation.TRUCE)) {
+
+                    if (!factionsList.containsKey(factionAt.getTag())) {
+                        factionsList.put(factionAt.getTag(), this.mapKeyChrs[Math.min(charIndex++, this.mapKeyChrs.length - 1)]);
                     }
+
+                    final char tag = factionsList.get(factionAt.getTag());
+                    row.then(String.valueOf(tag)).color(factionAt.getColorTo(senderFaction));
```

Comment:
psh we should just make it a string then, not a char. Edit my above comment to add a String.valueOf and make the map a Map<String, String>.